SPEC.md

```diff
@@ -0,0 +1,179 @@
+# libp2p Daemon
+
+The libp2p daemon is a standalone binary meant to make it easy to bring
+peer-to-peer networking to new languages without fully porting libp2p and all
+of its complexities.
+
+_At the moment, this is a living document. As such, it will be susceptible to
+changes until stabilization._
+
+## Structure
+
+### Overview
+
+There are two pieces to the libp2p daemon:
+
+- __Daemon__: A golang daemon that manages libp2p hosts and proxies streams to
+  the end user.
+- __Client__: A library written in any language that controls the daemon over
+  a protocol specified in this document, allowing end users to enjoy the
+  benefits of peer-to-peer networking without implementing a full libp2p stack.
+
+### Technical Details
+
+The libp2p daemon and client will communicate with each other over a simple Unix
+socket based protocol built with [protobuf](https://developers.google.com/protocol-buffers/).
+
+Future implementations may attempt to take advantage of shared memory (shmem)
+or other IPC constructs.
+
+## Protocol Specification
+
+### Data Types
+
+The data structures are defined in [pb/p2pd.proto](pb/p2pd.proto).
```

Comment:
Can we also specify the fact that the messages are varint-delimited?
Something like "All messages are varint-delimited."  should do.

Reply:
done :)